processor/src/operations/io_ops.rs

```diff
@@ -635,6 +635,36 @@ mod tests {
         assert_eq!(expected, process.stack.trace_state());
     }
 
+    /// Ensures that reading and writing in the same clock cycle results in an error.
+    #[test]
+    fn read_and_write_in_same_clock_cycle() {
```

Comment:
Should we also have a test to ensure that reading twice in the same clock cycle is fine?